### PR TITLE
protocols/autonat/src/protocol: Use ResponseStatus::from_i32

### PR DESCRIPTION
**Pull request by mxinden**
Would make the `.proto` file the single source of truth of the mapping of error codes to their names.

### CODE REVIEW

File: protocols/autonat/src/protocol.rs [hunk 1]
```diff
@@ -227,10 +229,12 @@ impl DialResponse {
 
         Ok(match msg.dial_response {
             Some(structs_proto::message::DialResponse {
-                status: Some(0),
+                status: Some(status),
                 status_text,
                 addr: Some(addr),
-            }) => {
+            }) if structs_proto::message::ResponseStatus::from_i32(status)
```
Comment (elenaf9) [inline protocols/autonat/src/protocol.rs:235]: Where does `from_i32` come from? It's not from `TryFrom<i32>`, right?

**Review by elenaf9 — APPROVED** [review]
Thanks!

File: protocols/autonat/src/protocol.rs (hunk 1 above)
Comment (mxinden) [reply protocols/autonat/src/protocol.rs:235]: Prost generates the following code:

``` rust
    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, ::prost::Enumeration)]
    #[repr(i32)]
    pub enum ResponseStatus {
        Ok = 0,
        EDialError = 100,
        EDialRefused = 101,
        EBadRequest = 200,
        EInternalError = 300,
    }
```

The `Enumeration` derive macro then generates the `from_i32` implementation:

https://docs.rs/prost-derive/latest/src/prost_derive/lib.rs.html#299-304

Comment (elenaf9) [reply protocols/autonat/src/protocol.rs:235]: Ahh, I didn't know that yet about prost. Thanks for the explanation!

**Pull request merged**